drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c

```diff
@@ -657,6 +652,21 @@ static void stop_streaming(struct vb2_queue *vq)
 			 ret);
 	}
 
+	/* wait for all buffers to be returned */
+	while (atomic_read(&port->buffers_with_vpu)) {
```

Comment:
Are atomic operations known to work on BCM283x?

Reply:
I believe they do, otherwise huge chunks of the kernel relying on it are going to fail.